lib/config/decrypt/gnupg.ts

```diff
@@ -0,0 +1,71 @@
+import os from 'node:os';
+import { isNonEmptyStringAndNotWhitespace } from '@sindresorhus/is';
+import { mkdtemp, outputFile, rm } from 'fs-extra';
+import upath from 'upath';
+import { logger } from '../../logger';
+import { exec } from '../../util/exec';
+
+const keyImported = new Set<string>();
+
+export async function tryDecryptGnupg(
+  privateKey: string,
+  encryptedStr: string,
+): Promise<string | null> {
+  const tmpDir = await mkdtemp(upath.join(os.tmpdir(), 'renovate-gpg-'));
+
+  if (!keyImported.has(privateKey)) {
+    try {
+      const keyFilePath = upath.join(tmpDir, 'key.pem');
+      await outputFile(keyFilePath, privateKey);
+      const { stdout, stderr } = await exec(
+        `gpg --batch --no-tty --yes --import ${keyFilePath}`,
```

Comment:
Please shlex:quote the value of encryptedFilePath to prevent arbitrary command injection

Reply:
How should that happen? `keyFilePath` is not dependend to userinput. it's a random file.

I'll quote it to be save against whitespaces on windows.